integration-tests/utils/integration-helpers.js

```diff
@@ -57,6 +57,21 @@ const createEntityInWorkspace = (page, billingProject, workspaceName, testEntity
   }, billingProject, workspaceName, testEntity)
 }
 
+const checkBucketAccess = async (page, billingProject, workspaceName, accessLevel = 'OWNER') => {
+  const details = await page.evaluate((billingProject, workspaceName) => {
+    return window.Ajax().Workspaces.workspace(billingProject, workspaceName).details()
+  }, billingProject, workspaceName)
+  const bucketName = details.workspace.bucketName
+  console.info(`Checking workspace access for ${billingProject}, ${workspaceName}, ${bucketName}.`)
+  // Try polling for workspace bucket access to be available.
+  await page.waitForFunction(async (billingProject, workspaceName, bucketName, accessLevel) => {
```

Comment:
Same question as above. I would really like to know whether these values are closed-over or not.

Reply:
It seemed like I needed to, but I would love it if I do not… I will experiment. 

What do you mean by closed-over?

Reply:
@dmohs Unfortunately without explicitly passing in the values, the code fails (variables aren't defined). I don't understand though why it is necessary.

Reply:
closed over = captured and made accessible in the scope of a closure